### PR TITLE
GuestFSModiDisk: let guestfs set variable backends.

### PR DESCRIPTION
**Pull request by sitoliu**
ID: 1543703
Signed-off-by: Sitong Liu <siliu@redhat.com>

### CODE REVIEW

**Comment by sitoliu** [comment]
1. smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.myprovider.timedrift_no_net.guest_pause_resume.clock_vm: PASS (127.15 s)
2. smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.myprovider.timedrift_no_net.guest_pause_resume.clock_host: PASS (128.50 s)
3. smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.myprovider.timerdevice.change_guest_clksource: PASS (196.55 s)

...
19:43:01 DEBUG| Launch the disk /home/kvm_autotest_root/images/rhel75-64-virtio-scsi.qcow2, wait...
19:43:05 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/rhel_dhcp-8-186/root' to '/'
19:43:05 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/sda1' to '/boot'
19:43:05 DEBUG| Umount all device partitions
19:43:05 WARNI| Update the first kernel entry to 'tsc' only
19:43:06 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/rhel_dhcp-8-186/root' to '/'
19:43:06 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/sda1' to '/boot'
19:43:06 DEBUG| Umount all device partitions
19:43:06 INFO | Context: Boot the guest
...

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by hereischen** [comment]
please remove '.' in your commit tittle

**Comment by sitoliu** [comment]
(1/1) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.ide.up.virtio_net.Guest.RHEL.7.5.x86_64.myprovider.migrate.with_pio_op.tcp: PASS (935.59 s)

.......
03:27:21 DEBUG| Launch the disk /home/kvm_autotest_root/images/rhel75-64-virtio.qcow2, wait...
03:27:24 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/rhel_dhcp-8-186/root' to '/'
03:27:25 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/sda1' to '/boot'
03:27:25 DEBUG| Umount all device partitions
03:27:26 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/rhel_dhcp-8-186/root' to '/'
03:27:26 INFO | Mount dev '/dev/rhel_dhcp-8-186/root' partitions '/dev/sda1' to '/boot'
03:27:26 DEBUG| Umount all device partitions
03:27:26 DEBUG| **Guest cmdline extra_params setting: added 'libata.dma=0'**
03:27:26 DEBUG| Generating random mac address for nic
03:27:26 DEBUG| Host does not support OpenVSwitch: Command 'ovs-vswitchd' could not be found in any of the PATH dirs: ['/bin', '/sbin', '/usr/bin', '/usr/local/sbin', '/usr/libexec', '/root/bin', '/usr/sbin', '/usr/local/bin']
03:27:26 DEBUG| Adding VM avocado-vt-vm1 NIC ifname t0-yg6Yxb to bridge switch
03:27:26 DEBUG| Host does not support OpenVSwitch: Command 'ovs-vswitchd' could not be found in any of the PATH dirs: ['/bin', '/sbin', '/usr/bin', '/usr/local/sbin', '/usr/libexec', '/root/bin', '/usr/sbin', '/usr/local/bin']
03:27:26 DEBUG| Could not find params qemu in /usr/share/avocado-plugins-vt/backends/qemu/qemu, searching the host PATH for one to use
03:27:26 DEBUG| Found /usr/libexec/qemu-kvm
......

**Comment by sitoliu** [comment]
Modified, thanks.

File: virttest/utils_disk.py [hunk 1]
```diff
@@ -445,7 +445,15 @@ class of guest disk using guestfs lib to do some operation(like read/write)
     on guest disk:
     """
 
-    def __init__(self, disk):
+    def __init__(self, disk, backend='direct'):
+        """
+        :params backend: Let libguestfs creats/connects to backend daemon
```
Comment (balamuruhans) [inline virttest/utils_disk.py]: 1. %s/creats/creates
2. can we add docstring for disk too

Comment (sitoliu) [reply virttest/utils_disk.py]: Modified it, thanks.

**Comment by balamuruhans** [comment]
Thanks @sitoliu @hereischen let's merge this PR.

**Pull request merged**